docs/databases/hadoop/install-and-configure-hadoop-cluster.md

```diff
@@ -0,0 +1,519 @@
+> The 3 nodes, *node-master*, *node1* and *node2* will be assumed to have IP addresses 192.0.2.1, 192.0.2.2 and 192.0.2.3, respectively. Change them accordingly to your own configuration.
+
+{: .note}
+>
+> This guide is written for a non-root user. Commands that require elevated privileges are prefixed with `sudo`. If you’re not familiar with the `sudo` command, see the [Users and Groups](/docs/tools-reference/linux-users-and-groups) guide. All commands in this guide are run with *hadoop* user if not specified otherwise.
+
+
+## Architecture of a Hadoop Cluster
+
+Before configuring the master and slave nodes, it's important to understand the different components of a Hadoop cluster:
+
+A **master node**  keeps knowledge about the distributed file system (like the `inode` table on an `ext3` filesystems) and schedules resources allocation. *Node-master* will handle this role in this guide, and host two daemons:
+
+*   The **NameNode**: manages the distributed file system and know where are stored data blocks inside the cluster
+*   The **ResourceManager**: manages the YARN jobs and take care of scheduling and execution on slave nodes
+
+**Slave nodes** store the actual data and provide processing power to run the jobs. They will be *node1* and *node2*, and will host two daemons:
+
+*   The **DataNode**: manages the actual data physically stored on the node and is called by the NameNode
+*   The **NodeManager**: manages execution of tasks on the node
+
+
+## Configure the System
+
+### Create Host File on Each Node
+
+For each node to be able to communicate with their names, edit the `/etc/hosts` file to add the IP address of the three servers. Don't forget to replace the sample IP with yours:
+
+  {: .file }
+  /etc/hosts
+  :   ~~~ conf
+      192.0.2.1    node-master
+      192.0.2.2    node1
+      192.0.2.3    node2
+      ~~~~
+
+### Distribute Authentication Key-pairs for the Hadoop User
+
+The master node will use ssh connections to other nodes with key-pair authentication to manage the cluster.
+
+1. Login to *node-master* as *hadoop* user, and generate a key:
+
+	   ssh-keygen -b 4096
+
+2. Then copy the key to other nodes. It is a good practice to also copy the key to *node-master* itself so that you can also use it as a DataNode if needed. Type the following commands, and enter hadoop user password when asked for. If you are prompted to add the key to known hosts, say "yes":
+
+        ssh-copy-id -i $HOME/.ssh/id_rsa.pub hadoop@node-master
+        ssh-copy-id -i $HOME/.ssh/id_rsa.pub hadoop@node1
+        ssh-copy-id -i $HOME/.ssh/id_rsa.pub hadoop@node2
+
+### Download and Unpack Hadoop Binaries
+
+Login to *node-master* as hadoop user, get the tarball of Hadoop from [Hadoop project page](https://hadoop.apache.org/) and unzip it (change the URL according to your nearest mirror):
+
+	cd
+	wget http://apache.mindstudios.com/hadoop/common/hadoop-2.8.1/hadoop-2.8.1.tar.gz
+	tar -xzf hadoop-2.8.1.tar.gz
+	ln -s hadoop-2.8.1 hadoop
```

Comment:
mv hadoop-2.8.1 hadoop

(symbolic link only add complexity, they are not needed)